src/main/java/com/dataguise/saas/util/Utility.java

```diff
@@ -0,0 +1,114 @@
+package com.dataguise.saas.util;
+
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import com.google.gson.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+
+public class Utility {
+    private static Logger logger = LoggerFactory.getLogger(Utility.class);
+
+    public static void runHealthChecks(String host, Integer port, String username, String password) throws InterruptedException {
+
+        ExponentialBackOff backoff = new ExponentialBackOff();
+        Boolean serverStatus = false;
+        Boolean dgSecureStatus = false;
+        String url = String.format("http://%s:%s/", host, port);
+        String loginUrl = String.format("http://%s:%s/dgcontroller/services/v1/sessions",host,port);
+
+        while(backoff.shouldRetry()) {
+            try {
+                int responseCode = checkDgSecureServerStatus(url);
+                if (200 <= responseCode && responseCode <= 399) {
+                    serverStatus = true;
+                }
+            } catch (IOException e) {
+                serverStatus = false;
+            }
+
+            if(serverStatus == true) {
+                try {
+                    int responseCode = checkDgSecureSetupStatus(loginUrl, username, password);
+                    if (200 <= responseCode && responseCode <= 399) {
+                        dgSecureStatus = true;
+                    }
+                } catch (IOException e) {
+                    dgSecureStatus = false;
+                }
+            }
+            if(serverStatus == true && dgSecureStatus == true) {
+                break;
+            }
+            else {
+                logger.info("Current backoff: "+backoff.getElapsedTime());
+                backoff.updateExponentially(backoff.getElapsedTime());
+            }
+        }
+
+        if(serverStatus == false) {
+            throw new InterruptedException(ErrorConstants.ERROR_CONNECTING_WITH_DGSECURE.getErrorMessage());
+        }else if (serverStatus == true && dgSecureStatus == false) {
+            throw new InterruptedException(ErrorConstants.LOGIN_FAIL_DGSECURE.getErrorMessage());
+        }
+
+    }
+
+
+    public static int checkDgSecureServerStatus(String url) throws IOException {
+        logger.info("Checking server status of DgSecure at " + url);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/util/Utility.java L70
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-167136428>*